### PR TITLE
MM-69516 Adding checks on post action response

### PR DESCRIPTION
**Pull request by avasconcelos114**
#### Summary

This PR adds some checks in the flow for post actions 

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-69516

## Change Impact: Medium 🟡

**Reasoning:** This adds authorization gating to user-facing post action handling and extends the shared `PluginAPI`/plugin utility surface, which can affect how requests succeed or fail. While tests were updated and new unit coverage was added for the permission helper, the change alters behavior on an authorization-critical path.

**Regression Risk:** Moderate—risk centers on edge-case authorization outcomes (e.g., users who previously reached the mutation/response path) and any missing/incorrect integration of the new `CanReadChannel` contract across callers. Updated tests reduce risk but do not eliminate it.

**QA Recommendation:** Perform targeted manual QA focused on post action endpoints (`accept`, `decline`, `tentative`, `respond`) for both authorized and unauthorized users, including verifying error/ephemeral messaging and response-update behavior. Minimal exploratory QA is recommended; avoid skipping manual QA.  
*Generated by CodeRabbitAI*

### CODE REVIEW

File: calendar/api/post_action.go [hunk 1]
```diff
@@ -102,6 +102,11 @@ func (api *api) postActionRespond(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	if !api.PluginAPI.CanReadChannel(p.ChannelId, user.MattermostUserID) {
+		utils.SlackAttachmentError(w, "Error: not authorized")
```
Comment (nang2049) [inline calendar/api/post_action.go]: By the time we reject as not authorized RespondToEvent has already mutated the event on MS Graph. An unauthorized user can still accept and tentatively respond to the meeting, we just refuse to render the updated post to them. Lets run the channel-read check before RespondToEvent.

**Review by nang2049 — COMMENTED** [review]
Thanks @avasconcelos114 can we verify my suggestion will improve the fix?

File: calendar/api/post_action.go (hunk 1 above)
Comment (avasconcelos114) [reply calendar/api/post_action.go]: Agreed! Changing that

**Review by edgarbellot — APPROVED** [review]
LGTM 🚀 thank you!

**Review by nang2049 — APPROVED** [review]
LGTM
